### PR TITLE
fix torrent name and site parsing

### LINKED ISSUE #62: [Bug] Website before rest of torrent name is included in title
Some of the torrent names failed to identify the site URL. it's included with the title. 

## Example 1: 
```
"www.1TamilMV.world - Raja Vikramarka (2024) Tamil HQ HDRip - 400MB - x264 - AAC - ESub.mkv"
```

Output:
```
{'quality': 'WEB-DL', 'year': 2024, 'codec': 'H.264', 'audio': 'AAC', 'filetype': 'MKV', 'language': 'Tamil', 'subtitles': 'English', 'size': '400MB', 'title': 'www.1TamilMV.world - Raja Vikramarka', 'encoder': 'HQ'}
```

Expected:
Site: 'www.1TamilMV.world'
title: 'Raja Vikramarka'

## Example 2:
```
"www.1TamilMV.world - Kotha Rangula Prapancham (2024) Telugu HQ PreDVD - 700MB - x264 - HQ Clean Aud.mkv"
```
 
Output:
```
{'year': 2024, 'codec': 'H.264', 'filetype': 'MKV', 'language': 'Telugu', 'size': '700MB', 'title': 'www.1TamilMV.world - Kotha Rangula Prapancham', 'encoder': 'Clean', 'site': 'Aud', 'excess': ['HQ', 'PreDVD', 'HQ']}
```

Expected Output:
Site: 'www.1TamilMV.world'
title: 'Kotha Rangula Prapancham'

### PR DESCRIPTION
**Pull request by mhdzumair**
Close #62

### CODE REVIEW

File: PTN/patterns.py [hunk 1]
```diff
@@ -291,7 +291,7 @@
     (r"\.?(iso)$", "ISO"),
 ]
 patterns["widescreen"] = "WS"
-patterns["site"] = [r"^(\[ ?([^\]]+?)\s?\])", r"^((?:www\.)?[\w-]+\.[\w]{2,4})\s+-\s*"]
+patterns["site"] = [r"^(www\.[\w-]+\.[\w-]+)\s+-\s*", r"^((?:www\.)?[\w-]+\.[\w-]+(?:\.[\w-]+)?)\s+-\s*", r"^(\[ ?([^\]]+?)\s?\])"]
```
Comment (platelminto) [inline PTN/patterns.py:303]: An explanation of the regex changes/additions would be highly appreciated! I'll still take a look manually, but it would help a lot.

It'd be even more appreciated if you could add that explanation as a comment surrounding the regexes themselves - these are dense, and getting denser. A short sentence or two (if possible) would help a lot, now and for the future.

File: PTN/post.py [hunk 2]
```diff
@@ -76,7 +76,7 @@ def try_encoder_before_site(self, unmatched):
                     "site",
                     (match_s + len(encoder_raw), match_e),
                     self._clean_string(site_raw),
-                    overwrite=True,
+                    overwrite=False,
```
Comment (platelminto) [inline PTN/post.py:79]: Why was this necessary?

**Review by platelminto — COMMENTED** [review]
Thanks for the PR!

I'm trying to actually use GitHub features lol, hopefully this review stuff works ok. My comments are mostly stuff that'd help me out, and aren't mandatory - I'll still be taking a look manually, and then likely approve. Addressing the comments would make that process faster though!

File: PTN/post.py (hunk 2 above)
Comment (mhdzumair) [reply PTN/post.py:79]: Sorry, i forgot to reply for this.  This change needed for title 

`"www.1TamilMV.world - Kotha Rangula Prapancham (2024) Telugu HQ PreDVD - 700MB - x264 - HQ Clean Aud.mkv"` 

If you try to debug this title, it was correctly identified the site as `www.1TamilMV.world` then due to this try_encoder_before_site function, it's overriding the previously found correct site into `'Aud'` which is not correct.

File: PTN/patterns.py (hunk 1 above)
Comment (mhdzumair) [reply PTN/patterns.py:303]: Added the description on the change in the order

**Comment by mhdzumair** [comment]
@platelminto apart from this, Any design decision for making this library support to other language? I'm looking forward to adding support to other language but haven't thought of any plan yet.

**Comment by mhdzumair** [comment]
@platelminto any plan merging this?

**Comment by platelminto** [comment]
Sorry it took a while! Been very busy with school.

For other languages, sticking to just Python for now - I don't have the time for another language rewrite. You are welcome to start one though!

**Comment by mhdzumair** [comment]
> For other languages, sticking to just Python for now

I mean not the other program language. I mean to add support for multi language torrent name.

example: arabic, Chinese etc.
